From 244c22bcbff750ced06c03e333f0cf902ad28a7e Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Thu, 28 Mar 2024 15:10:48 +0530
Subject: [PATCH] arm64: dts: st: Enable Superspeed support for USB3DR in
 stm32mp257f-dk

Add the required node(s) for USB3-DWC3 Superspeed support : COMBOPHY
Add combo-phy handle in USB3DR node
Enable usb-role-switch on stm32mp257f-dk, to support dual role USB2
with power delivery on Type-C connector.
There's a dependency on the type-c connector controller, to enable
and manage the super speed switch (orientation).

Change-Id: Ic556b8effb9854b50b8793f9572f544c704c721a
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/374829
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 27 +++++++++++++++++++----
 1 file changed, 23 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -193,6 +193,11 @@
 	status = "okay";
 };
 
+&combophy {
+	st,ssc-on;
+	status = "okay";
+};
+
 &crc {
 	status = "okay";
 };
@@ -559,7 +564,17 @@
 			compatible = "st,stm32mp25-typec";
 			reg = <0x35>;
 			interrupts-extended = <&intc_rpmsg 0>;
-			status = "disabled";
+			status = "okay";
+			connector {
+				compatible = "usb-c-connector";
+				label = "USB-C";
+
+				port {
+					typec_ep: endpoint {
+						remote-endpoint = <&dwc3_ep>;
+					};
+				};
+			};
 		};
 	};
 };
@@ -728,9 +743,13 @@
 	status = "okay";
 
 	dwc3: usb@48300000 {
-		maximum-speed = "high-speed";
-		role-switch-default-mode = "peripheral";
+		phys            = <&usb2_phy2>, <&combophy PHY_TYPE_USB3>;
+		phy-names       = "usb2-phy", "usb3-phy";
 		usb-role-switch;
-		status = "disabled";
+		port {
+			dwc3_ep: endpoint {
+				remote-endpoint = <&typec_ep>;
+			};
+		};
 	};
 };
